### ISSUE TITLE
Improve S1264: Update RSPEC to include using initialization for simple assignation

### ISSUE THREAD

**Issue by sebastien-marichal** [SonarSource/sonar-dotnet#8249]
From this [community post](https://community.sonarsource.com/t/s1264-replace-for-with-while-false-positive/98118).

Update S1264 rspec to include the following snippet as a non-compliant example:

```cs
int i;

for (i = 0; i < indexes.Count - 1;)
{ }
```

A more complex example:

```cs
int i;
int j = 3;
for (i = 0, Console.WriteLine($"Start: i={i}, j={j}"); i < j; i++, j--, Console.WriteLine($"Step: i={i}, j={j}"))
{
    //...
}
```

In other words, anything else than a declaration in the initializer section should be considered as non-compliant.
